dwave/optimization/model.pyx

```diff
@@ -845,53 +846,72 @@ cdef class Model:
     def to_networkx(self):
         """Convert the model to a NetworkX graph.
 
-        Note:
-            Currently requires the installation of a GNU compiler.
-
         Returns:
-            A :obj:`NetworkX <networkx:networkx.Graph>` graph.
+            A :obj:`NetworkX <networkx:networkx.MultiDiGraph>` graph.
 
         Examples:
             This example converts a model to a graph.
 
             >>> from dwave.optimization.model import Model
             >>> model = Model()
-            >>> c = model.constant(8)
-            >>> i = model.integer((20, 30))
-            >>> g = model.to_networkx()   # doctest: +SKIP
-        """
-        # Todo: adapt to use iter_symbols()
-        # This whole method will need a re-write, it currently only works with gcc
-        # but it is useful for development
+            >>> one = model.constant(1)
+            >>> two = model.constant(2)
+            >>> i = model.integer()
+            >>> model.minimize(two * i - one)
+            >>> G = model.to_networkx()
 
-        import re
-        import networkx
+            One advantage of converting to NetworkX is the wide availability
+            of drawing tools. See NetworkX's
+            `drawing <https://networkx.org/documentation/stable/reference/drawing.html>`_
+            documentation.
 
-        G = networkx.DiGraph()
+            This example uses a `DAGVIZ <https://wimyedema.github.io/dagviz/>`_ to
+            draw the NetworkX graph created in the example above.
 
-        cdef cppNode* ptr
-        for i in range(self._graph.num_nodes()):
-            ptr = self._graph.nodes()[i].get()
+            >>> import dagviz                      # doctest: +SKIP
+            >>> r = dagviz.render_svg(G)           # doctest: +SKIP
+            >>> with open("model.svg", "w") as f:  # doctest: +SKIP
+            ...     f.write(r)
 
-            # this regex is compiler specific! Don't do this for the general case
-            match = re.search("\d+([a-zA-z]+Node)", str(typeid(deref(ptr)).name()))
-            if not match:
-                raise ValueError
+            This creates the following image:
 
-            u = (match[1], <long>(ptr))
+            .. figure:: /_images/to_networkx_example.svg
+               :width: 500 px
+               :name: to-networkx-example
```

Comment:
Would you mind renaming to something like `nlModelToNetworkxExample` for the figure's label and something like  `nl_model_to_networkx_example` for the filename above? Once the package is updated in the SDK, having package-specific names is helpful for preventing label conflicts.

Reply:
How about
```suggestion
               :name: dwave-optimization-to-networkx-example
```

Reply:
works for me, thanks!